src/main/java/com/aws/greengrass/util/platforms/unix/LinuxPlatform.java

```diff
@@ -0,0 +1,143 @@
+                }
+                if (!Files.exists(cg.getSubsystemComponentPath(componentName))) {
+                    Files.createDirectory(cg.getSubsystemComponentPath(componentName));
+                }
+            }
+        } catch (IOException e) {
+            logger.atError().setCause(e).kv("componentName", componentName)
+                    .log("Failed to create system resource manager");
+        } catch (InterruptedException e) {
+            logger.atWarn("Got interrupted");
+        }
+    }
+
+    public void applyResourceLimits(String componentName, Map<String, Object> resourceLimit) {
+        try {
+            if (resourceLimit.containsKey(MEMORY_KEY)) {
+                long memoryLimitInKB = (long) resourceLimit.get(MEMORY_KEY);
+
+                // TODO: add input validation
+                String memoryLimit = Long.toString(memoryLimitInKB * 1024);
+                Files.write(LinuxCgroup.Memory.getSubsystemComponentPath(componentName).resolve(CGROUP_MEMORY_LIMITS),
+                        memoryLimit.getBytes());
+            }
+
+            if (resourceLimit.containsKey(CPU_KEY)) {
+                double cpu = (double) resourceLimit.get(CPU_KEY);
+
+                byte[] content = Files.readAllBytes(
+                        LinuxCgroup.CPU.getSubsystemComponentPath(componentName).resolve("cpu.cfs_period_us"));
+                int cpuPeriodUs = Integer.parseInt(new String(content));
+
+                int cpuQuotaUs = (int) (cpuPeriodUs * cpu);
+                String cpuQuotaUsStr = Integer.toString(cpuQuotaUs);
+
+                Files.write(LinuxCgroup.CPU.getSubsystemComponentPath(componentName).resolve("cpu.cfs_quota_us"),
+                        cpuQuotaUsStr.getBytes());
+            }
+
+        } catch (IOException e) {
+            logger.atError().setCause(e).kv("componentName", componentName)
+                    .log("Failed to apply resource limits");
+        }
+    }
+
+    public void addProcessToResourceManager(String componentName, Process process) {
+        if (process == null) {
+            logger.atError().log("The process doesn't exist");
+            return;
+        }
+        int pid = PidUtil.getPid(process);
+        try {
+            Files.write(LinuxCgroup.Memory.getSubsystemComponentPath(componentName).resolve("cgroup.procs"),
+                    Integer.toString(pid).getBytes());
+            Files.write(LinuxCgroup.CPU.getSubsystemComponentPath(componentName).resolve("cgroup.procs"), Integer
+                    .toString(pid).getBytes());
+        } catch (IOException e) {
+            logger.atError().setCause(e).kv("componentName", componentName).kv("processId", pid)
+                    .log("Failed to add pid to the cgroup");
+        }
+    }
+
+    private void runCmdWithRoot(String command) throws IOException, InterruptedException {
+        String[] cmd = getUserDecorator().withUser(getPrivilegedUser()).decorate(command);
```

Comment:
Does this still decorate when we're already root? Because we don't want it to do that. `Exec` handles that properly, not sure if calling these directly also covers it.

Reply:
Why not use some relevant utility from Exec.java here? Also, why is it `getPrivilegedUser ` instead of the current user that's running nuclues?